cmd/minishift/state/instance_dirs.go

```diff
@@ -17,42 +17,46 @@ limitations under the License.
 package state
 
 import (
-	"github.com/minishift/minishift/pkg/minikube/constants"
 	"path/filepath"
+
+	"github.com/minishift/minishift/pkg/minikube/constants"
 )
 
 type MinishiftDirs struct {
-	Home       string
-	Config     string
-	Machines   string
-	Certs      string
-	Cache      string
-	IsoCache   string
-	OcCache    string
-	ImageCache string
-	Addons     string
-	Logs       string
-	Tmp        string
+	Home         string
+	Config       string
+	GlobalConfig string
+	Machines     string
+	Certs        string
+	Cache        string
+	IsoCache     string
+	OcCache      string
+	ImageCache   string
+	Addons       string
+	Logs         string
+	Tmp          string
 }
 
 var InstanceDirs *MinishiftDirs
 
 // Constructor for MinishiftDirs
 func NewMinishiftDirs(baseDir string) *MinishiftDirs {
+	minishiftHomeDir := constants.GetMinishiftHomeDir()
 	// We use a global cache, sharing the cache directory of the default 'minishift' instance
-	cacheDir := filepath.Join(constants.GetMinishiftHomeDir(), "cache")
+	cacheDir := filepath.Join(minishiftHomeDir, "cache")
 
 	return &MinishiftDirs{
-		Home:       baseDir,
-		Certs:      filepath.Join(baseDir, "certs"),
-		Machines:   filepath.Join(baseDir, "machines"),
-		Addons:     filepath.Join(baseDir, "addons"),
-		Logs:       filepath.Join(baseDir, "logs"),
-		Tmp:        filepath.Join(baseDir, "tmp"),
-		Config:     filepath.Join(baseDir, "config"),
-		Cache:      cacheDir,
-		IsoCache:   filepath.Join(cacheDir, "iso"),
-		OcCache:    filepath.Join(cacheDir, "oc"),
-		ImageCache: filepath.Join(cacheDir, "images"),
+		Home:         baseDir,
+		Certs:        filepath.Join(baseDir, "certs"),
+		Machines:     filepath.Join(baseDir, "machines"),
+		Addons:       filepath.Join(baseDir, "addons"),
+		Logs:         filepath.Join(baseDir, "logs"),
+		Tmp:          filepath.Join(baseDir, "tmp"),
+		Config:       filepath.Join(baseDir, "config"),
+		GlobalConfig: filepath.Join(minishiftHomeDir, "config"),
```

Comment:
For default profile GlobalConfig and Config would be same but that is not an issue as https://github.com/minishift/minishift/blob/master/cmd/minishift/cmd/root.go#L434-L443 `os.MkdirAll` does not do anything if the directory is present.